cnd/src/asset/ethereum/mod.rs

```diff
@@ -0,0 +1,34 @@
+use num::bigint::ParseBigIntError;
+
+mod erc20;
+mod ether;
+
+pub use self::{
+    erc20::{Erc20, Erc20Quantity},
+    ether::Ether,
+};
+
+pub trait FromWei<W> {
+    fn from_wei(wei: W) -> Self;
+}
+
+pub trait TryFromWei<W>
+where
+    Self: std::marker::Sized,
+{
+    fn try_from_wei(wei: W) -> Result<Self, Error>;
+}
+
+#[derive(Clone, Debug, thiserror::Error, PartialEq)]
+pub enum Error {
+    #[error("value provided overflows")]
+    Overflow,
+    #[error("parsing error encountered")]
+    Parse(ParseBigIntError),
```

Comment:
This will generate you the `From` impl below :)

```suggestion
    Parse(#[from] ParseBigIntError),
```